hal/network/lwip/wiznet/wiznetif_config.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2023 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "wiznetif_config.h"
+
+#include "logging.h"
+LOG_SOURCE_CATEGORY("net.en.cfg")
+
+#include "system_error.h"
+
+#include "check.h"
+#include "delay_hal.h"
+
+#if HAL_PLATFORM_NCP
+#include "system_cache.h"
+#endif // HAL_PLATFORM_NCP
```

Comment:
Whats the point of this check? I think the only platform that doesnt have `HAL_PLATFORM_NCP` is the gcc platform, which I dont think should be building the wiznet stuff?

Reply:
I borrowed it from R510 FOTA stuff, which has Unit Tests... which needed that.  It probably doesn't hurt to leave it?